chrome/content/zotero/preferences/preferences_advanced.js

```diff
@@ -719,11 +721,12 @@ Zotero_Preferences.Attachment_Base_Directory = {
 		// Prune preferences of any libraries that no longer exist
 		var existentLibraryIDs = libraries.map(l => parseInt(l.libraryID));
 		var savedLibraryIDs =
-				Object.keys(
-					JSON.parse(
-						Zotero.Prefs.get("librarySaveRelativeAttachmentPaths") || "{}"
-					)
-				).map(id => parseInt(id));
+			Object.keys(
+				JSON.parse(
+					Zotero.Prefs.get("librarySaveRelativeAttachmentPaths") || "{}"
+				)
+			).map(id => parseInt(id));
+
 		savedLibraryIDs.forEach(function (libraryID) {
 			if (existentLibraryIDs.indexOf(libraryID) == -1) {
 				Zotero.debug(`Pruning attachment base path preferences for non-existent library '${libraryID}'`);
```

Comment:
Hmm...  Should `clearPath` be called on every `libraryID` that's being pruned?  Or is it safe to assume that all attachments have been removed already, so there wouldn't be any attachment paths that need to be updated?
Is it possible for any attachments to have a `libraryID` of a library that `Zotero.Libraries.getAll()` doesn't return?